megamek/src/megamek/common/verifier/TestSmallCraft.java

```diff
@@ -436,9 +446,15 @@ public double getWeightAmmo() {
     @Override
     public double getWeightMisc() {
         double weight = 0.0;
+        boolean NC3 = hasNC3(smallCraft);
         // Add in extra fire control system weight for exceeding base slot limit
         for (double extra : extraSlotCost(smallCraft)) {
-            weight += extra;
+            //Naval C3 doubles the weight of FCS, if present
+            if (NC3 == true) {
```

Comment:
Rather than doubling it during final weight calculations it should be doubled when the extra tonnage per arc is calculated. This way requires it to be doubled everywhere it is used which makes for more fragile code.